File: FE/src/services/admin/searchAnalytics.service.ts
import apiClient from '@/services/api/api'
import { PREFIX_API_ADMIN } from '@/constants/url'

export interface SearchKeywordResponse {
  keyword: string
  searchCount: number
}

export interface SearchTrendResponse {
  date: string
  searchCount: number
}

export interface SearchQueryData {
  id: string
  queryText: string
  resultCount: number
  hasResults: boolean
  category: string
  createdAt: number
}

const searchAnalyticsService = {
  getTopKeywords: async (
    period: string = 'week',
    limit: number = 20
  ): Promise<SearchKeywordResponse[]> => {
    const response = await apiClient.get(`${PREFIX_API_ADMIN}/analytics/search/top-keywords`, {
      params: { period, limit },
    })
    return response.data
  },

  getNoResultQueries: async (period: string = 'week'): Promise<SearchQueryData[]> => {
    const response = await apiClient.get(`${PREFIX_API_ADMIN}/analytics/search/no-results`, {
      params: { period },
    })
    return response.data
  },

  getSearchTrends: async (period: string = 'week'): Promise<SearchTrendResponse[]> => {
    const response = await apiClient.get(`${PREFIX_API_ADMIN}/analytics/search/trends`, {
      params: { period },
    })
    return response.data
  },
}

export default searchAnalyticsService

File: BE/src/main/java/udpm/hn/server/core/admin/analytics/service/SearchAnalyticsService.java
package udpm.hn.server.core.admin.analytics.service;

import udpm.hn.server.core.admin.analytics.model.response.SearchKeywordResponse;
import udpm.hn.server.core.admin.analytics.model.response.SearchTrendResponse;
import udpm.hn.server.entity.SearchQuery;

import java.util.List;

public interface SearchAnalyticsService {

    /**
     * Get top search keywords within a period
     * 
     * @param period "day", "week", or "month"
     * @param limit  number of results
     */
    List<SearchKeywordResponse> getTopKeywords(String period, int limit);

    /**
     * Get searches that returned no results
     * 
     * @param period "day", "week", or "month"
     */
    List<SearchQuery> getNoResultQueries(String period);

    /**
     * Get search count trends over time
     * 
     * @param period "week" or "month"
     */
    List<SearchTrendResponse> getSearchTrends(String period);
}
